shm/player.hpp

```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <memory>
+
+#include "cargo.hpp"
+#include "ship.hpp"
+
+class Player {
+public:
+    size_t getSpeed() const { return ship_->getSpeed(); }
+    Cargo* getCargo(size_t index) const;
+
+private:
+    std::shared_ptr<Ship> ship_;
```

Comment:
What about unique pointer? (Based on readme.md)

Reply:
I have updated it in new commit